app/src/main/java/org/oppia/app/topic/play/StorySummaryAdapter.kt

```diff
@@ -109,27 +113,81 @@ class StorySummaryAdapter(
 
       val chapterList = storySummaryViewModel.storySummary.chapterList
       binding.chapterRecyclerView.adapter = ChapterSummaryAdapter(chapterList, chapterSummarySelector)
+      if (currentExpandedChapterListIndex != null && currentExpandedChapterListIndex == position) {
+        val rotate = AnimationUtils.loadAnimation(context, R.anim.rotation_clockwise_180)
+        binding.chapterListDropDownIcon.startAnimation(rotate)
+        expand(binding.chapterListContainer)
+      } else {
+        val rotate = AnimationUtils.loadAnimation(context, R.anim.rotation_anti_clockwise_180)
+        binding.chapterListDropDownIcon.startAnimation(rotate)
+        collapse(binding.chapterListContainer)
+      }
 
       binding.root.setOnClickListener {
-        val previousIndex: Int? = currentExpandedChapterListIndex
-        currentExpandedChapterListIndex =
-          if (currentExpandedChapterListIndex != null && currentExpandedChapterListIndex == position) {
-            null
-          } else {
-            position
-          }
-        expandedChapterListIndexListener.onExpandListIconClicked(currentExpandedChapterListIndex)
-        if (previousIndex != null && currentExpandedChapterListIndex != null && previousIndex == currentExpandedChapterListIndex) {
-          notifyItemChanged(currentExpandedChapterListIndex!!)
+        val previousItem = currentExpandedChapterListIndex
+        if (binding.chapterListContainer.isVisible) {
+          currentExpandedChapterListIndex = null
         } else {
-          if (previousIndex != null) {
-            notifyItemChanged(previousIndex)
+          currentExpandedChapterListIndex = position
+        }
+        expandedChapterListIndexListener.onExpandListIconClicked(currentExpandedChapterListIndex)
+        notifyItemChanged(position)
+        if (previousItem != null && previousItem != position) {
+          notifyItemChanged(previousItem)
+        }
+      }
+    }
+
+    private fun expand(chapterListContainer: View) {
+      chapterListContainer.clearAnimation()
+      val matchParentMeasureSpec =
+        View.MeasureSpec.makeMeasureSpec((chapterListContainer.parent as View).width, View.MeasureSpec.EXACTLY)
+      val wrapContentMeasureSpec = View.MeasureSpec.makeMeasureSpec(0, View.MeasureSpec.UNSPECIFIED)
+      chapterListContainer.measure(matchParentMeasureSpec, wrapContentMeasureSpec)
+      val targetHeight = chapterListContainer.measuredHeight
+
+      // Older versions of android (pre API 21) cancel animations for views with a height of 0.
+      chapterListContainer.layoutParams.height = 1
+      chapterListContainer.visibility = View.VISIBLE
+      val expandAnimation = object : Animation() {
+        override fun applyTransformation(interpolatedTime: Float, t: Transformation) {
+          if (interpolatedTime == 1f) {
```

Comment:
We shouldn't ever do direct equality with floats due to precision error. Suggest instead using Float.approximatelyEquals (in our own FloatExtensions file).

Reply:
Are you referring to something like this: https://alvinalexander.com/source-code/java-approximately-equals-method-function

Also, what exactly do you mean by `our own FloatExtensions file`, do we need to create a file, which extends float and in which there is a function which does the part of `approximatelyEquals` ?